tools/config.py

```diff
@@ -797,6 +798,13 @@ def validate_config(self):
         return True
 
 
+    @property
+    def name(self):
+        if "artifact_name" in self.app_config_data:
```

Comment:
Not completely familiar with this file yet.... will this work for `mbed_lib.json` as well as `mbed_app.json`?